### PR TITLE
[READY][MOSTLY NON-MODULAR] Flavour-ifies conflict resolution messages, adds mold to it

### PR DESCRIPTION
**Pull request by FlamingLily**
## About The Pull Request
Adds some flavour to the shuttle lockdown caused by blob/nukies/revs/gangs/etc. The announcement now states *why* the shuttle is locked down "Biohazard, imminent destruction, police lockdown, "kill the revs", narsie rises"

Also adds mold to the "pending conflict resolution" shuttle lockdown, and adds an announcement to it

Removes prison from the mold's available spawn locations

## How This Contributes To The Skyrat Roleplay Experience
Being able to leave a station with a severe biohazard is dumb, and if it's that bad you're going to evacuate, you should adhere to the atmosphere of a deserted frontier station and evacuate everyone to lavaland before detonating the nuke to prevent the spread of the biohazard. The round would end either way, one makes more sense from an IC POV.

## Changelog
:cl:
tweak: "Conflict Resolution" messages now detail the cause of the conflict
balance: Molds are now biohazards, and make an announcement accordingly.
balance: Molds can no longer spawn in the prison
/:cl:

### CODE REVIEW

**Comment by GoldenAlpharex** [comment]
My issue with this is, a mold or a spacevine that spawns in a hidden room in maints and remains self-contained will just delay the round for no good reason.
Also, if you're going to do this, I hope that it does not include crew-made space vines, because that means any antag with access to Kudzu could stop the shuttle from leaving indefinitely, until it's dealt with.

**Comment by SpaceVampire** [comment]
This probably could have been split up into multiple PRs, and vines and mold are really not a biohazard level threat. It's also just an unfun way of artificially extending a round that most people would rather just finish. I only see this causing issues, and honestly given you can just fly away to the interlink at anytime, we should probably stop pretending that lockdowns make much sense at all.

**Comment by FlamingLily** [comment]
> My issue with this is, a mold or a spacevine that spawns in a hidden room in maints and remains self-contained will just delay the round for no good reason.
> Also, if you're going to do this, I hope that it does not include crew-made space vines, because that means any antag with access to Kudzu could stop the shuttle from leaving indefinitely, until it's dealt with.

mm, fair point, maybe I'll look into "if there's over X amount of vines"

**Comment by FlamingLily** [comment]
> This probably could have been split up into multiple PRs, and vines and mold are really not a biohazard level threat. It's also just an unfun way of artificially extending a round that most people would rather just finish. I only see this causing issues, and honestly given you can just fly away to the interlink at anytime, we should probably stop pretending that lockdowns make much sense at all.

Or maybe it's the interlink that makes no sense

**Comment by FlamingLily** [comment]
indentation :pensive:

File: modular_skyrat/master_files/code/controllers/subsystem/shuttle.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/datum/controller/subsystem/shuttle/
+	var/hostileEnvironmentTypes = list()
+
+/datum/controller/subsystem/shuttle/proc/registerHostileEnvironment(datum/bad, reason)
+	hostileEnvironments[bad] = TRUE
+	hostileEnvironmentTypes += reason
+	checkHostileEnvironment()
+
+/datum/controller/subsystem/shuttle/proc/checkHostileEnvironment()
+	for(var/datum/d in hostileEnvironments)
```
Comment (tf-4) [inline modular_skyrat/master_files/code/controllers/subsystem/shuttle.dm]: https://github.com/tgstation/tgstation/pull/56872

Comment (FlamingLily) [reply modular_skyrat/master_files/code/controllers/subsystem/shuttle.dm]: o yeah fair point

File: modular_skyrat/master_files/code/controllers/subsystem/shuttle.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		if(!istype(d) || QDELETED(d))
+			hostileEnvironments -= d
```
Comment (FlamingLily) [inline modular_skyrat/master_files/code/controllers/subsystem/shuttle.dm]:
```suggestion
	for(var/datum/environment in hostileEnvironments)
		if(!istype(environment) || QDELETED(environment))
			hostileEnvironments -= environment
```

**Comment by hugmeorelse** [comment]
1. Sure. More Flavor is not bad.
2. No. Round delay over those two events sound awful.
3. Sure.

**Comment by FlamingLily** [comment]
> 1. Sure. More Flavor is not bad.
> 2. No. Round delay over those two events sound awful.
> 3. Sure.

Vines actually doesn't do it anymore, I forgot to remove it from the PR.

As for mold, it won't delay the round, if you can't contain it, don't bother calling the emergency shuttle, nuke the station to prevent escape of a lethal biohazard, the round would end either way, one just being a far more RP-friendly way to do it

**Comment by FlamingLily** [comment]
However, it's suddenly occuring to me that mold doesn't have an announcement anymore, so I'll be updating that

**Comment by FlamingLily** [comment]
I'm not developing for this server anymore, Xyel can suck my dick.

If you want me to change, Xyel, maybe start listening to my complaints and stop hurling abuse at me over DMs, and more importantly, saying "I'm the cause of all the Dev team leaving" when that's proveably not true.

**Pull request closed without merging**